browser/ui/webui/brave_new_tab_ui.h

```diff
@@ -0,0 +1,27 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_WEBUI_BRAVE_NEW_TAB_UI_H_
+#define BRAVE_BROWSER_UI_WEBUI_BRAVE_NEW_TAB_UI_H_
+
+#include "brave/browser/ui/webui/basic_ui.h"
+
+class PrefChangeRegistrar;
+
+class BraveNewTabUI : public BasicUI {
+ public:
+  explicit BraveNewTabUI(content::WebUI* web_ui, const std::string& host);
+  ~BraveNewTabUI() override;
+
+ private:
+  void CustomizeNewTabWebUIProperties();
+  void RenderFrameCreated(content::RenderFrameHost* render_frame_host) override;
+  void OnPreferenceChanged();
+
+  std::unique_ptr<PrefChangeRegistrar> pref_change_registrar_;
```

Comment:
If the lifetime of object member is same with this class, do we have to use |unique_ptr|?

Reply:
per above, same as for reasons of forward declaring only. Will leave it.

Reply:
Ok